tests/tests.rs

```diff
@@ -555,6 +555,15 @@ fn test_follow() {
     );
 }
 
+#[test]
+fn test_follow_broken_symlink() {
+    let mut te = TestEnv::new(DEFAULT_DIRS, DEFAULT_FILES);
+    te.create_broken_symlink("broken_symlink")
+        .expect("Failed to create broken symlink.");
+
+    te.assert_output(&["--follow", "--type", "f", "symlink"], "broken_symlink");
```

Comment:
I think this test actually shows that something is not yet correct in the implementation.

I would *not* expect `--type file` to show a broken symlink. A broken symlink should only show up for `--type symlink` (or if no `--type` is specified).

`--follow` shouldn't have any influence on the result.

I guess I would expect this behavior:

``` rust
te.assert_output(&["symlink"], "broken_symlink");
te.assert_output(&["--type", "link", "symlink"], "broken_symlink");

te.assert_output(&["--type", "file", "symlink"], "");

te.assert_output(&["--follow", "--type", "link", "symlink"], "broken_symlink");
te.assert_output(&["--follow", "--type", "file", "symlink"], "");
```